gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
The reason that json unmarshalling happens twice is because for certain values of certain types, they're automatically marshelled to string representations which aren't and should not be surrounded by square brackets. For example:

Let's say you're making a request, and one field on that request is a repeated series of ints. If you don't include anything in that field, then when it is marshalled into a string, it won't result in a string of `[]`, it will instead be marshalled into a string of `null`.

That's why I chose to implement this by attempting decoding twice, instead of via blind string manipulation. If you guys think there's a better way to do it, I do want to know.